### PR TITLE
Add tf.multi_one_hot that one-hot encodes multiple columns of Tensor

### PR DESCRIPTION
**Pull request by mkurovski**
Hello,
here is the PR for my feature request #16044 
Excited to see your feedback for my first TF PR and hope to give something useful back to this great library ;)

### CODE REVIEW

**Comment by squall-1002** [comment]
Adapted GitHub email address to align with the one I signed the CLA with.

**Comment by squall-1002** [comment]
I signed the CLA and now have the same email address in GitHub with which I once signed it

**Comment by squall-1002** [comment]
I signed the CLA

**Comment by rmlarsen** [comment]
@squall-1002 Are you sure you signed with the same email you used to submit the PR?

**Comment by rmlarsen** [comment]
Maybe you have to say "I signed it!"

**Comment by squall-1002** [comment]
I signed it!

**Comment by rmlarsen** [comment]
@squall-1002 Please resolve the conflicts by rebasing.

**Comment by rmlarsen** [comment]
@caisq can you review this, please?

**Comment by rmlarsen** [comment]
@caisq ping

**Review by caisq — CHANGES_REQUESTED** [review]
Thanks for the pull request, @squall-1002 
Sorry about the delay in reviewing the PR.
Can you please add unit tests for the new method?

**Comment by squall-1002** [comment]
@caisq No worries, it also took me some time to get back and add some unit tests - looking forward to your response.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @rmlarsen: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by rmlarsen** [comment]
@caisq one more look?

File: tensorflow/python/ops/array_ops.py
Comment (caisq) [inline tensorflow/python/ops/array_ops.py]: Do you want to make this into actually exectuable code? This code block does say three-backticks-python.

File: tensorflow/python/ops/array_ops.py
Comment (caisq) [inline tensorflow/python/ops/array_ops.py]: So just to make sure I understand: in the case where `depth_list` is length-1, this reduces to normal one-hot, right?

File: tensorflow/python/ops/array_ops.py
Comment (caisq) [inline tensorflow/python/ops/array_ops.py]: Can you describe the default behavior?

File: tensorflow/python/ops/array_ops.py
Comment (caisq) [inline tensorflow/python/ops/array_ops.py]: Can you describe the expected shape and dtype of `indices`?

File: tensorflow/python/ops/array_ops.py
Comment (caisq) [inline tensorflow/python/ops/array_ops.py]: This block of code can be simplified with list comprehension. Something like:

tensor_list = [one_hot(indices[:, col], depth_list[col], on_value_list[col], off_value_list[col]) for col in range(n_features)]

**Review by caisq — CHANGES_REQUESTED** [review]

**Comment by rmlarsen** [comment]
@caisq thanks for the review

File: tensorflow/python/ops/array_ops.py
Comment (squall-1002) [reply tensorflow/python/ops/array_ops.py]: Yes, I took the doctest format from the other docstrings, but in fact some of the code in between wasn't executable. I fixed this and reworked the example ;)

File: tensorflow/python/ops/array_ops.py
Comment (squall-1002) [reply tensorflow/python/ops/array_ops.py]: If the length of `depth_list` is 1, we face a regular one-hot-encoding, if this is what you mean.

File: tensorflow/python/ops/array_ops.py
Comment (squall-1002) [reply tensorflow/python/ops/array_ops.py]: I added the default behavior for `on_values_list` and `off_values_list`.

File: tensorflow/python/ops/array_ops.py
Comment (squall-1002) [reply tensorflow/python/ops/array_ops.py]: I added the expected shape of indices `indices` being basically a matrix of several instances and columns representing features that we like to one-hot-encode and concatenate within multi-one-hot.

File: tensorflow/python/ops/array_ops.py
Comment (squall-1002) [reply tensorflow/python/ops/array_ops.py]: Good point!
Thanks for the example, I simplified it accordingly.

**Comment by rmlarsen** [comment]
@caisq another look, please?

**Review by caisq — DISMISSED** [review]

**Comment by jhseu** [comment]
Mind also updating the API goldens? See the test failures.

**Comment by squall-1002** [comment]
Hi @jhseu, I am about to try it and calling the `api_compatibility_test.py`with the `--update_goldens`flag set to `True`, but it fails due to the missing `api_objects_pb2`module which I also cannot find on the most recent public repository. I was also searching for the problem, but wasn't successful. Am I doing something wrong?
`Traceback (most recent call last):
  File "api_compatibility_test.py", line 47, in <module>
    from tensorflow.tools.api.lib import api_objects_pb2
ImportError: No module named 'tensorflow.tools.api'`

**Comment by jhseu** [comment]
@gunan knows more, but I believe it might only work with Python 2 and Linux.

**Comment by squall-1002** [comment]
@jhseu I updated the API goldens, can you trigger the tests again, please?

**Comment by jhseu** [comment]
Test failures are unrelated. Just waiting on API review.

**Comment by squall-1002** [comment]
@caisq already approved the changes, or does it need a new approval for merge?

**Comment by caisq** [comment]
@jhseu Not sure how API review is supposed to work on github nowadays, but should any API reviewers be added to this PR?

**Comment by jhseu** [comment]
Usually API reviewers meet once weekly and go over the PRs. If another week goes by, I'll ping someone manually.

**Comment by martinwicke** [comment]
(for API review)

We looked at this and we have some trouble with the semantics of this. This seems to be very hard to use correctly, and we have the concern that it is easier to write your own (and possibly specialized) version of this rather than understand what exactly `multi_one_hot` does. Therefore, we would rather not add `multi_one_hot` to the API. I will close this PR.

I apologize for the back and forth -- we should have made this decision on the original issue, and not make you do a bunch of work. We'll make some changes in order catch this type of issue earlier in the future.

**Pull request closed without merging**
